Imagine/Filter/Loader/ResampleFilterLoader.php

```diff
@@ -0,0 +1,160 @@
+     * @param array          $options
+     *
+     * @throws LoadFilterException
+     *
+     * @return ImageInterface
+     */
+    public function load(ImageInterface $image, array $options = array())
+    {
+        $options = $this->resolveOptions($options);
+        $tmpFile = $this->getTemporaryFile($options['temp_dir']);
+
+        try {
+            $image->save($tmpFile, $this->getImagineSaveOptions($options));
+            $image = $this->imagine->open($tmpFile);
+            $this->delTemporaryFile($tmpFile);
+        } catch (\Exception $exception) {
+            $this->delTemporaryFile($tmpFile);
+            throw new LoadFilterException('Unable to save/open file in resample filter loader.', null, $exception);
+        }
+
+        return $image;
+    }
+
+    /**
+     * @param string $path
+     *
+     * @throws \RuntimeException
+     *
+     * @return string
+     */
+    private function getTemporaryFile($path)
+    {
+        if (!is_dir($path) || false === $file = tempnam($path, 'liip-imagine-bundle')) {
+            throw new \RuntimeException(sprintf('Unable to create temporary file in "%s" base path.', $path));
+        }
+
+        return $file;
+    }
+
+    /**
+     * @param $file
+     *
+     * @throws \RuntimeException
+     */
+    private function delTemporaryFile($file)
+    {
+        if (file_exists($file)) {
+            unlink($file);
+        }
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    private function getImagineSaveOptions(array $options)
+    {
+        $saveOptions = array(
+            'resolution-units' => $options['unit'],
+            'resolution-x' => $options['x'],
+            'resolution-y' => $options['y'],
+        );
+
+        if (isset($options['filter'])) {
+            $saveOptions['resampling-filter'] = $options['filter'];
+        }
+
+        return $saveOptions;
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    private function resolveOptions(array $options)
+    {
+        $resolver = new OptionsResolver();
+
+        $resolver->setRequired(array('x', 'y', 'unit', 'temp_dir'));
+        $resolver->setDefined(array('filter'));
+        $resolver->setDefault('temp_dir', sys_get_temp_dir());
+        $resolver->setDefault('filter', 'UNDEFINED');
+
+        $resolver->setAllowedTypes('x', array('int'));
+        $resolver->setAllowedTypes('y', array('int'));
+        $resolver->setAllowedTypes('temp_dir', array('string'));
+        $resolver->setAllowedTypes('filter', array('string'));
+
+        $resolver->setAllowedValues('unit', array(
+            ImageInterface::RESOLUTION_PIXELSPERINCH,
+            ImageInterface::RESOLUTION_PIXELSPERCENTIMETER
+        ));
+
+        $resolver->setNormalizer('filter', function (Options $options, $value) {
+            foreach (array('\Imagine\Image\ImageInterface::FILTER_%s', '\Imagine\Image\ImageInterface::%s', '\%s', '%s') as $format) {
```

Comment:
I added them as a very "micro-optimization" as it does remove some opcodes (otherwise it first searches in the defined namespace, followed by the global namespace, the latter of which we want to force for the first two formats, but now that you bring this up, we could likely change:
```php
array('\Imagine\Image\ImageInterface::FILTER_%s', '\Imagine\Image\ImageInterface::%s', '\%s', '%s')
```
to
```php
array('\Imagine\Image\ImageInterface::FILTER_%s', '\Imagine\Image\ImageInterface::%s', '%s')
```

Of course, we don't *need* them anywhere, but I think it adds clarity and acts as a small speedup.